backend/tournesol/views/inconsistencies.py

```diff
@@ -8,12 +8,188 @@
+                "comparison__entity_1__uid",
+                "comparison__entity_2__uid",
+                "criteria",
+                "score",
+            )
+        )
+
+        for self.criterion in poll.criterias_list:
+
+            self._fill_graph_parameters(comparisons)
+
+            cycles_count, comparison_trios_count = self._count_cycles_and_comparison_trios()
+
+            stats[self.criterion] = {
+                "cycles_count": cycles_count,
+                "comparison_trios_count": comparison_trios_count,
+            }
+
+        response = {
+            "count": len(self.cycles),
+            "results": self.cycles,
+            "stats": stats,
+        }
+
+        return Response(Length3CyclesSerializer(response).data)
+
+    def _fill_graph_parameters(self, comparisons: list):
+        """
+        Fill the set containing all the entities.
+
+        The incoming and outgoing connections sets are used to search the cycles.
+        all_connections is only used to count all the comparison trios.
+
+        outgoing_connections: Dictionary of sets for each entity.
+                              Each element in the sets is a preferred entity.
+        incoming_connections: Idem but each element in the sets is a worse rated entity.
+
+        all_connections: all the edges (also including comparisons with a score of 0 as edges)
+        """
+        self.all_entities = set()
+        self.outgoing_connections = {}
+        self.incoming_connections = {}
+
+        # all the edges (also including comparisons with a score of 0 as edges)
+        self.all_connections = {}
+
+        criterion_comparisons = list(
+            filter(lambda d: d["criteria"] == self.criterion, comparisons)
+        )
+
+        for comparison in criterion_comparisons:
+            entity_1 = comparison["comparison__entity_1__uid"]
+            entity_2 = comparison["comparison__entity_2__uid"]
+
+            if entity_1 not in self.all_entities:
+                self.all_entities.add(entity_1)
+                self.incoming_connections[entity_1] = set()
+                self.outgoing_connections[entity_1] = set()
+                self.all_connections[entity_1] = set()
+
+            if entity_2 not in self.all_entities:
+                self.all_entities.add(entity_2)
+                self.incoming_connections[entity_2] = set()
+                self.outgoing_connections[entity_2] = set()
+                self.all_connections[entity_2] = set()
+
+            if comparison["score"] > 0:
+                # Entity_2 is preferred
+                self.incoming_connections[entity_2].add(entity_1)
+                self.outgoing_connections[entity_1].add(entity_2)
+
+            elif comparison["score"] < 0:
+                # Entity_1 is preferred
+                self.incoming_connections[entity_1].add(entity_2)
+                self.outgoing_connections[entity_2].add(entity_1)
+            else:
+                continue
```

Comment:
I shouldn't have put an "else: continue" there. all_connections is used for comparison trios, for which the score doesn't matter.